### PR TITLE
Selenium runner: Ability to configure using command-line arguments

### PR DESCRIPTION
**Pull request by alandipert**
Previously various bits of Selenium configuration were embedded in the Selenium runner code.

This PR converts the runner to Kotlin and make Selenium fully user-configurable by exposing all arguments to Selenium as command-line options.

Running `java -jar selenium-1.0.0-jar-with-dependencies.jar` with no arguments prints usage info and an example of running Chrome headlessly.

Note that the shinyjster-specific XPath selector (`"//span[@id='shinyjster_progress_val'`) is no longer embedded and must be provided as the `XPATH` positional parameter.

### CODE REVIEW

File: inst/selenium/src/main/kotlin/com/rstudio/seleniumRunner/Main.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import io.github.bonigarcia.wdm.DriverManagerType
+import io.github.bonigarcia.wdm.WebDriverManager
+import org.openqa.selenium.By
+import org.openqa.selenium.Dimension
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.support.ui.ExpectedConditions
+import org.openqa.selenium.support.ui.WebDriverWait
+import kotlin.system.exitProcess
+
+val types = enumValues<DriverManagerType>().map { it.name.toLowerCase() }.joinToString(",")
```
Comment (schloerke) [inline inst/selenium/src/main/kotlin/com/rstudio/seleniumRunner/Main.kt]: Can you add a comment on the existing types?

File: inst/selenium/src/main/kotlin/com/rstudio/seleniumRunner/Main.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    exitProcess(0)
+}
+
+fun driverOptions(driverName: String, args: List<String>): Any? {
+    if (args.isEmpty()) return null
+    val className = when(driverName) {
+        "chrome" -> "org.openqa.selenium.chrome.ChromeOptions"
+        "firefox" -> "org.openqa.selenium.firefox.FirefoxOptions"
+        "opera" -> "org.openqa.selenium.opera.OperaOptions"
+        "iexplorer" -> "org.openqa.selenium.ie.InternetExplorerOptions"
```
Comment (schloerke) [inline inst/selenium/src/main/kotlin/com/rstudio/seleniumRunner/Main.kt]: Edge does not support options?

**Comment by schloerke** [comment]
Does it compile the same with `mvn package` then move the file?

File: inst/selenium/src/main/kotlin/com/rstudio/seleniumRunner/Main.kt (hunk 2 above)
Comment (alandipert) [reply inst/selenium/src/main/kotlin/com/rstudio/seleniumRunner/Main.kt]: I don't know for sure, but I couldn't find anything like "EdgeOptions" in the Selenium package. phantomjs, selenenium_server_standalone (don't know what that is...) and chromium are the other drivers that didn't seem to have Options classes.

**Comment by alandipert** [comment]
> Does it compile the same with `mvn package` then move the file?

Right, `mvn package` still builds and the output file has the same name so I didn't have to change the GH workflow.

File: inst/selenium/src/main/kotlin/com/rstudio/seleniumRunner/Main.kt (hunk 2 above)
Comment (schloerke) [reply inst/selenium/src/main/kotlin/com/rstudio/seleniumRunner/Main.kt]: `import org.openqa.selenium.edge.EdgeOptions;`

It doesn't support much of anything, but it'd complete the set.

**Pull request merged**